internal/zinc-core/src/main/java/xsbti/compile/AnalysisStore.java

```diff
@@ -0,0 +1,37 @@
+package xsbti.compile;
+
+import java.util.Optional;
+
+/**
+ * Defines a store interface that provides analysis read and write capabilities to users.
+ *
+ * The store is a backend-independent interface that allows implementors to decide how
+ * the analysis stores are read and written before or after every incremental compile.
+ */
+public interface AnalysisStore {
```

Comment:
As far as I can tell, the impact to sbt/sbt is around the relatively small changes around the storage.
Do you think you can split it up into another PR along with the ripple PRs?